lib/util/update-metadata-builder.js

```diff
@@ -0,0 +1,149 @@
+	this.body = [];
+}
+
+/**
+ * Insert add operation to the request body.
+ *
+ * @param {string} path - path designates the key. it must be prefixed with /.
+ * @param {object} value - the value could any one of the types string, enum, float, and date (RFC 3339).
+ * @returns {void}
+ */
+UpdateMetadataBuilder.prototype.insertAdd = function(path, value) {
+	this.insertPath('add', path, value);
+};
+
+/**
+ * Insert replace operation to the request body.
+ *
+ * @param {string} path - path designates the key. it must be prefixed with /.
+ * @param {object} value - the value could any one of the types string, enum, float, and date (RFC 3339).
+ * @returns {void}
+ */
+UpdateMetadataBuilder.prototype.insertReplace = function(path, value) {
+	this.insertPath('replace', path, value);
+};
+
+/**
+ * Insert remove operation to the request body.
+ *
+ * @param {string} path - path designates the key. it must be prefixed with /.
+ * @returns {void}
+ */
+UpdateMetadataBuilder.prototype.insertRemove = function(path) {
+	assert(typeof path === 'string' && path.length > 0, 'path designates the key. It is mandatory.');
+
+	var item = {
+		op: 'remove',
+		path: path
+	};
+	this.body.push(item);
+};
+
+/**
+ * Insert test operation to the request body.
+ *
+ * @param {string} path - path designates the key. it must be prefixed with /.
+ * @param {object} value - the value could any one of the types string, enum, float, and date (RFC 3339).
+ * @returns {void}
+ */
+UpdateMetadataBuilder.prototype.insertTest = function(path, value) {
+	this.insertPath('test', path, value);
+};
+
+/**
+ * Insert move operation to the request body.
+ *
+ * @param {string} from - from designates the source key. it must be prefixed with /.
+ * @param {object} value - the value could any one of the types string, enum, float, and date (RFC 3339).
+ * @returns {void}
+ */
+UpdateMetadataBuilder.prototype.insertMove = function(from, value) {
+	this.insertFrom('move', from, value);
+};
+
+/**
+ * Insert copy operation to the request body.
+ *
+ * @param {string} from - from designates the source key. it must be prefixed with /.
+ * @param {object} value - the value could any one of the types string, enum, float, and date (RFC 3339).
+ * @returns {void}
+ */
+UpdateMetadataBuilder.prototype.insertCopy = function(from, value) {
+	this.insertFrom('copy', from, value);
+};
+
+/**
+ * Insert path based operation in to the request body.
+ *
+ * @param {String} op - The possible operations are add, replace, test.
+ * @param {string} path - path designates the key. it must be prefixed with /.
+ * @param {object} value - the value could any one of the types string, enum, float, and date (RFC 3339).
+ * @returns {void}
+ */
+UpdateMetadataBuilder.prototype.insertPath = function(op, path, value) {
+	assert(typeof op === 'string' && op.length > 0, 'op designates the operation. It is mandatory.');
+	assert(typeof path === 'string' && path.length > 0, 'path designates the key. It is mandatory.');
```

Comment:
While you're checking this, it's probably a good idea to ensure the path starts with '/'